process/block/postprocess/intermediateResults.go

```diff
@@ -258,7 +258,8 @@ func (irp *intermediateResultsProcessor) checkSmartContractResultIntegrity(scr *
 		return process.ErrNilTxHash
 	}
 	sndId := irp.shardCoordinator.ComputeId(scr.SndAddr)
-	if !core.IsEmptyAddress(scr.SndAddr) && sndId != irp.shardCoordinator.SelfId() {
+	dstId := irp.shardCoordinator.ComputeId(scr.RcvAddr)
```

Comment:
What happens if scr.SndAddr or scr.RcvAddr are nil. Maybe you should move this inside
```
if !core.IsEmptyAddress(scr.SndAddr) && !core.IsEmptyAddress(scr.RcvAddr) {
   sndShardID := irp.shardCoordinator.ComputeId(scr.SndAddr)
   dstShardID := irp.shardCoordinator.ComputeId(scr.RcvAddr)
   if sndShardID != irp.shardCoordinator.SelfId() && dstShardID != irp.shardCoordinator.SelfId() {
      return process.ErrShardIdMissmatch
   }
}
```

Reply:
nil checks are done above. for both sender, receiver and prevtxhash

Reply:
modified to your suggestion